service/src/intTest/java/uk/nhs/adaptors/gp2gp/common/storageconnectors/AzureStorageConnectorTest.java

```diff
@@ -0,0 +1,65 @@
+package uk.nhs.adaptors.gp2gp.common.storageconnectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class AzureStorageConnectorTest {
```

Comment:
There should be a single integration test that uses the factory or, better yet, dependency injection, to get an instance of StorageConnector. Which backend it tests against should depend entirely upon the configuration used when running the tests. Using the mock/local should be the default, but it should be possible to set environment variables that would cause it to run the AWS or Azure code instead.